### PR TITLE
Use mixedCase for public functions/variables in examples/docs

### PR DESCRIPTION
**Pull request by Dan-Nolan**
### - What I did

Update Vyper examples/documentation so public functions/variables use mixedCase instead of snake_case. 

Stems from discussion in #905 

### - How I did it

Meticulous find & replace while running tests 😄 

### - How to verify it

- Running `python setup.py test` on the examples folder shows the examples are working
- Looking over docs to ensure all references have been updated

### - Description for the changelog

None.

### - Cute Animal Picture

image

### CODE REVIEW

File: examples/name_registry.vy [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-@public
-def register(name: bytes[100], owner: address):
-    assert self.registry[name] == ZERO_ADDRESS  # check name has not been set yet.
-    self.registry[name] = owner
-
-
-@public
-@constant
-def lookup(name: bytes[100]) -> address:
-    return self.registry[name]
```
Comment (Dan-Nolan) [inline examples/name_registry.vy]: This is a duplicate file, the test uses `examples/name_registry/name_registry.vy` which is more consistent with the file structure.

File: examples/market_maker/on_chain_market_maker.vy [hunk 2]
```diff
@@ -7,21 +7,21 @@ totalTokenQty: public(uint256(currency_value))
...
-token_address: address(ERC20)
+tokenAddress: address(ERC20)
 owner: public(address)
 
 # Sets the on chain market maker with its owner, intial token quantity,
 # and initial ether quantity
 @public
 @payable
-def initiate(token_addr: address, token_quantity: uint256(currency_value)):
+def initiate(tokenAddr: address, tokenQuantity: uint256(currency_value)):
```
Comment (jacqueswww) [inline examples/market_maker/on_chain_market_maker.vy]: @Dan-Nolan @fubuloubu I think we should keep arguments and local variables as is, camelCase for those too are a bit too extreme in my opinion.

Comment (Dan-Nolan) [reply examples/market_maker/on_chain_market_maker.vy]: Sounds good, I'll revert the latest commit

File: docs/vyper-by-example.rst [hunk 3]
```diff
@@ -397,24 +397,24 @@ by the contract creator, we have access to the contract creator’s address with
...
+``giveRightToVote()``, which we will explore next. We loop through the two
 proposals from the argument and insert them into ``proposals`` mapping with
 their respective index in the original array as its key.
 
 Now that the initial setup is done, lets take a look at the functionality.
 
 .. literalinclude:: ../examples/voting/ballot.vy
   :language: python
-  :pyobject: give_right_to_vote
+  :pyobject: giveRightToVote
```
[the remark this answers is not in the document]
Comment (jacqueswww) [reply docs/vyper-by-example.rst]: strange indeed will have to take a look, not cool :P

**Pull request merged**